src/olympia/addons/tests/test_models.py

```diff
@@ -1362,16 +1363,43 @@ def test_binary_components_property(self):
         file.update(binary_components=True)
         assert addon.binary_components
 
-    def test_is_incomplete(self):
-        addon = Addon.objects.get(pk=3615)
-        SubmitStep.objects.create(addon=addon, step=6)
+    def test_listed_is_incomplete_no_categories(self):
+        addon = Addon.objects.get(id=3615)
+        assert not addon.is_incomplete()  # Confirm not incomplete already.
+
+        addon.categories.all().delete()
+        addon = Addon.objects.get(id=3615)
         assert addon.is_incomplete()
 
-    def test_unlisted_is_incomplete(self):
-        addon = Addon.objects.get(pk=3615)
-        SubmitStep.objects.create(addon=addon, step=2)
+    def test_listed_is_incomplete_no_summary(self):
+        addon = Addon.objects.get(id=3615)
+        assert not addon.is_incomplete()  # Confirm not incomplete already.
+
+        delete_translation(addon, 'summary')
+        addon = Addon.objects.get(id=3615)
         assert addon.is_incomplete()
 
+    def test_listed_is_incomplete_no_license(self):
+        addon = Addon.objects.get(id=3615)
+        assert not addon.is_incomplete()  # Confirm not incomplete already.
+
+        addon.latest_version.update(license=None)
+        addon = Addon.objects.get(id=3615)
+        assert addon.is_incomplete()
+
+    def test_unlisted_is_incomplete(self):
+        addon = Addon.objects.get(id=3615)
+        addon.update(is_listed=False)
+        addon.current_version.update(channel=amo.RELEASE_CHANNEL_UNLISTED)
+        assert not addon.is_incomplete()  # Confirm not incomplete already.
+
+        # Clear everything
+        addon.all_categories = []
+        delete_translation(addon, 'summary')
+        addon.latest_version.update(license=None)
+        addon = Addon.with_unlisted.get(id=3615)
+        assert not addon.is_incomplete()  # Still not incomplete
+
```

Comment:
Maybe then update it back to `is_listed=True` and re-check `is_incomplete()` ? Or would that be too much ? Up to you.